### ISSUE TITLE
Provider crash

### ISSUE THREAD

**Issue by Vasanth2896** [hashicorp/terraform-provider-google#12235]
### Affected Resource(s)

* google_*

Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If this is a bug report (for when something is there, but doesn't work how it should), please use the [bug issue template](https://github.com/terraform-providers/terraform-provider-google/issues/new?template=bug.md) instead.

If this is an enhancement / feature request (for when something (a resource, field, etc.) is missing, and should be added), please use the [enhancement template](https://github.com/terraform-providers/terraform-provider-google/issues/new?template=enhancement.md) instead.

If you have a support request or question, please check out the other resources listed [here](https://github.com/terraform-providers/terraform-provider-google/issues/new?template=question.md).

If your issue does not fit into one of the above templates, please fill out the following:

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment or [link](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) the pull request to this issue.

### Terraform Version

Terraform v.1.2.5

### Affected Resource(s)

* google_4.3.1

### Terraform Configuration Files (if applicable)

```
#Bric ID: BRIC-176
#Bric Name:gcp_cloudsql
#Version: v1
#Description: Create a new GCP Cloud SQL
#Compatibility : Tested on terraform v1.0.7
#Created on: 24-Nov-2021.
#Created by: Uma.

variable "project" {}
variable "credentials_file" {}
variable "region" {}
variable "name" {}
variable "database_version" {}
variable "root_password" {}
variable "tier" {}
variable "availability_type" {}
variable "disk_type" {}
variable "disk_size" {}
variable "disk_autoresize" {}
variable "zone" {}
variable "binary_log_enabled" {}
variable "enabled" {}
variable "ipv4_enabled" {}
variable "vpc" {}
#variable "security_token" {}


provider "google" {
  project     = var.project
  credentials = file(var.credentials_file)
  region      = var.region
  #token = var.security_token

}


resource "google_sql_database_instance" "cloudsql" {
  name             = var.name
  root_password    = var.root_password
  database_version = var.database_version

  settings {
    availability_type = var.availability_type
    tier              = var.tier

    disk_type = var.disk_type

    disk_size       = var.disk_size
    disk_autoresize = var.disk_autoresize //by default it will be true

    location_preference {
      zone = var.zone
    }
    backup_configuration {

      binary_log_enabled = var.binary_log_enabled
      enabled            = var.enabled
    }
    ip_configuration {

      ipv4_enabled    = true
      private_network = "projects/${var.project}/global/networks/${var.vpc}"
    }
  }
}
```

### Issue Description

Not able to Initialize the provider. The Provider crashes

output:
```
Error: Plugin did not respond
│
│   with provider["registry.terraform.io/hashicorp/google"],
│   on bric_176_gcp_cloudsql.tf line 28, in provider "google":
│   28: provider "google" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ConfigureProvider call. The
│ plugin logs may contain more details.
╵

Stack trace from the terraform-provider-google_v4.31.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbbc285]

goroutine 84 [running]:
golang.org/x/oauth2/authhandler.authHandlerSource.Token(0x3879cf0, 0xc000d58400, 0xc0009ca150, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        golang.org/x/oauth2@v0.0.0-20220524215830-622c5d57e401/authhandler/authhandler.go:48 +0x65
golang.org/x/oauth2.(*reuseTokenSource).Token(0xc0000b15c0, 0x0, 0x0, 0x0)
        golang.org/x/oauth2@v0.0.0-20220524215830-622c5d57e401/oauth2.go:304 +0xaf
golang.org/x/oauth2.(*Transport).RoundTrip(0xc0000b15e0, 0xc0002c8200, 0x0, 0x0, 0x0)
        golang.org/x/oauth2@v0.0.0-20220524215830-622c5d57e401/transport.go:45 +0x9e
net/http.send(0xc0002c8200, 0x38391c0, 0xc0000b15e0, 0x0, 0x0, 0x0, 0xc0006e9f98, 0x418377, 0x1, 0x0)
        net/http/client.go:251 +0x454
net/http.(*Client).send(0xc0009edb60, 0xc0002c8200, 0x0, 0x0, 0x0, 0xc0006e9f98, 0x0, 0x1, 0xc0000b1620)
        net/http/client.go:175 +0xff
net/http.(*Client).do(0xc0009edb60, 0xc0002c8200, 0x0, 0x0, 0x0)
        net/http/client.go:717 +0x45f
net/http.(*Client).Do(...)
        net/http/client.go:585
github.com/hashicorp/terraform-provider-google/google.sendRequestWithTimeout.func1(0xc0009bea60, 0xc0009c2900)
        github.com/hashicorp/terraform-provider-google/google/transport.go:83 +0x25b
github.com/hashicorp/terraform-provider-google/google.retryTimeDuration.func1(0x2)
        github.com/hashicorp/terraform-provider-google/google/retry_utils.go:21 +0x32
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/resource/wait.go:27 +0x5b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1(0xc0009be8a0, 0xc0009ca1c0, 0xc0006fa2a0, 0xc000d59c00, 0xc0009b0418, 0xc0009b0410)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/resource/state.go:110 +0x2e9
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/resource/state.go:83 +0x1c6

Error: The terraform-provider-google_v4.31.0_x5 plugin crashed!
```

**Comment by megan07** [comment]
Hi @Vasanth2896! Sorry you're running into this issue. Would you mind sharing any debug logs we could see? Thanks!

**Comment by philippeboyd** [comment]
@megan07 I get a similar error with the latest `v4.42` reverting back to `v4.41` works

```
Stack trace from the terraform-provider-google_v4.42.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x24dcb65]

goroutine 750 [running]:
github.com/hashicorp/terraform-provider-google/google.flattenBucketLifecycleRuleCondition(0xc0002dc0e0)
	github.com/hashicorp/terraform-provider-google/google/resource_storage_bucket.go:1083 +0x45
github.com/hashicorp/terraform-provider-google/google.flattenBucketLifecycle(0xc001066f00)
	github.com/hashicorp/terraform-provider-google/google/resource_storage_bucket.go:1067 +0x276
github.com/hashicorp/terraform-provider-google/google.setStorageBucket(0x2d59a8b?, 0x42?, 0xc000656e00, {0xc0010edca0, 0x1a}, {0xc000054150, 0x65})
	github.com/hashicorp/terraform-provider-google/google/resource_storage_bucket.go:1493 +0xf51
github.com/hashicorp/terraform-provider-google/google.resourceStorageBucketRead(0x7f4e18b15f08?, {0x2cccd40?, 0xc00036f000})
	github.com/hashicorp/terraform-provider-google/google/resource_storage_bucket.go:732 +0x3a7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x3232400?, {0x3232400?, 0xc00101c390?}, 0xd?, {0x2cccd40?, 0xc00036f000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00098e2a0, {0x3232400, 0xc00101c390}, 0xc000ed8ea0, {0x2cccd40, 0xc00036f000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00065e948, {0x3232400?, 0xc00101c2d0?}, 0xc000e90e00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:613 +0x497
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0005620a0, {0x3232400?, 0xc000863170?}, 0xc000f67740)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:747 +0x41e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x2c5f7e0?, 0xc0005620a0}, {0x3232400, 0xc000863170}, 0xc0012bf110, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000001e0, {0x32377e8, 0xc000b6e820}, 0xc000e2fd40, 0xc0002f0fc0, 0x4387e30, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc0000001e0, {0x32377e8, 0xc000b6e820}, 0xc000e2fd40, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

Error: The terraform-provider-google_v4.42.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

Operation failed: failed running terraform plan (exit 1)
```

**Comment by philippeboyd** [comment]
Newly released [v4.42.1](https://github.com/hashicorp/terraform-provider-google/releases/tag/v4.42.1) fixes the issue 👏

**Comment by megan07** [comment]
Hi @Vasanth2896, is this still an issue for you?

**Comment by Vasanth2896** [comment]
Hi Megan,

Its been a long time since I have tried it.
I am not sure whether the issue still occurs,

Thanks,
Vasanth

**Comment by shahasim** [comment]
Hi,

I have had a similar issue with version v4.53.0_x5 plugin, please let me know if you need anything else from me.

I am trying to run the plan command using our in-house `make` script. I have downgraded to v 4.50.0 and I don't get the error anymore.  

```
╷
│ Error: Plugin did not respond
│ 
│   with provider["registry.terraform.io/hashicorp/google"],
│   on __env.tf line 20, in provider "google":
│   20: provider "google" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-google_v4.53.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x5ecc01]

goroutine 28 [running]:
github.com/hashicorp/terraform-plugin-log/internal/logging.GetProviderRootLogger(...)
        github.com/hashicorp/terraform-plugin-log@v0.7.0/internal/logging/provider.go:12
github.com/hashicorp/terraform-plugin-log/tflog.Info({0x0, 0x0}, {0x30d2361, 0x85}, {0x0, 0x0, 0x0})
        github.com/hashicorp/terraform-plugin-log@v0.7.0/tflog/provider.go:74 +0x61
github.com/hashicorp/terraform-provider-google/google.getCurrUserEmail(0xc0000e9500, {0xc000098300?, 0x356c360?}, 0xc000a46750)
        github.com/hashicorp/terraform-provider-google/google/framework_utils.go:44 +0x25f
github.com/hashicorp/terraform-provider-google/google.(*frameworkProvider).logGoogleIdentities(_, {_, _}, {{0x2, {0xc000044103, 0x2f}}, {0x0, {0x0, 0x0}}, {0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-provider-google/google/framework_config.go:1219 +0x698
github.com/hashicorp/terraform-provider-google/google.(*frameworkProvider).SetupClient(_, {_, _}, {{0x2, {0xc000044103, 0x2f}}, {0x0, {0x0, 0x0}}, {0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-provider-google/google/framework_config.go:1149 +0x28f
github.com/hashicorp/terraform-provider-google/google.(*frameworkProvider).ConfigureWithData(_, {_, _}, {{0x2, {0xc000044103, 0x2f}}, {0x0, {0x0, 0x0}}, {0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-provider-google/google/framework_config.go:47 +0x23e
github.com/hashicorp/terraform-provider-google/google.(*frameworkProvider).Configure(0xc0000e9500, {0x3575238, 0xc000ca8d80}, {{0xc00156a710, 0x5}, {{{0x357b990, 0xc000ccbf50}, {0x2b3a1e0, 0xc000cc0a50}}, {0x357ce58, ...}}}, ...)
        github.com/hashicorp/terraform-provider-google/google/framework_provider.go:821 +0x225
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ConfigureProvider(0xc0000ca000, {0x3575238, 0xc000ca8d80}, 0xc000a467c0, 0xc000a46740)
        github.com/hashicorp/terraform-plugin-framework@v1.1.1/internal/fwserver/server_configureprovider.go:15 +0x10f
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ConfigureProvider(0xc0000ca000, {0x3575238?, 0xc000ca8930?}, 0x3?)
        github.com/hashicorp/terraform-plugin-framework@v1.1.1/internal/proto5server/server_configureprovider.go:36 +0x292
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ConfigureProvider({0xc0009a6a50, 0xc0009a6ab0, {0xc000fb2ba0, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-plugin-mux@v0.8.0/tf5muxserver/mux_server_ConfigureProvider.go:25 +0x1b3
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0xc00075aaa0, {0x3575238?, 0xc000ca53b0?}, 0xc000a461c0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:556 +0x2ce
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0x2f08860?, 0xc00075aaa0}, {0x3575238, 0xc000ca53b0}, 0xc00057f030, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:331 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00010e3c0, {0x357bc20, 0xc00103ab60}, 0xc000ecdd40, 0xc0010275f0, 0x47d66b8, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc00010e3c0, {0x357bc20, 0xc00103ab60}, 0xc000ecdd40, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-google_v4.53.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

make: *** [plan] Error 1
(base) 
```

**Comment by rileykarson** [comment]
@shahasim: That crash is https://github.com/hashicorp/terraform-provider-google/issues/13731, which should be resolved by `4.53.1`.

I see that this thread has collected a few different crashes, and doesn't have a clear reproduction to be able to move forward with. I'm gonna close it as wontfix. If any earlier crashes are reproducible on current provider versions, please open new issues!
